app/controllers/clearance/passwords_controller.rb

```diff
@@ -96,7 +101,16 @@ def forbid_missing_token
   end
 
   def forbid_non_existent_user
-    unless find_user_by_id_and_confirmation_token
+    unless find_user_from_password_reset
+      flash_failure_when_forbidden
+      render template: 'passwords/new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.